lib/scrollyteller/scrollyactions.ts

```diff
@@ -0,0 +1,279 @@
+  if (value === 'false') return false;
+  return fallback;
+}
+
+// Optional helper to fetch your real chart API. Adjust to your actual structure.
+function getApiFromContext(ctx: ActionContext): ParaChartsApi | null {
+  const parachartAny = ctx.parachart as any;
+  const api = parachartAny?.api;
+
+  if (!api || typeof api.getChart !== 'function') {
+    // eslint-disable-next-line no-console
+    console.warn('[scrollyActions] ctx.parachart.api missing or invalid');
+    return null;
+  }
+
+  return api as ParaChartsApi;
+}
+
+// ─────────────────────────────────────────────────────────────────────────────
+// Default scrollyActions
+// ─────────────────────────────────────────────────────────────────────────────
+
+export const scrollyActions: ActionMap = {
+  /**
+   * Highlight one or more series on the chart.
+   *
+   * Examples:
+   *   data-para-enter="highlightSeries(series='revenue')"
+   *   data-para-enter="highlightSeries(revenue, expenses)"
+   */
+  highlightSeries: (ctx: ActionContext, ...args: ParaActionArg[]) => {
+    // console.warn('[scrollyActions.highlightSeries]', ctx, args);
+
+    const named = getNamedArgs(args);
+    const positional = getPositionalArgs(args);
+
+    const primarySeries = asString(named.series);
+    const seriesIds: string[] = [];
+
+    if (primarySeries) {
+      seriesIds.push(primarySeries);
+    }
+    for (const value of positional) {
+      const id = asString(value);
+      if (id) seriesIds.push(id);
+    }
+
+    if (!ctx.chartId || seriesIds.length === 0) return;
+
+    const emphasis = asNumber(named.emphasis, 1);
+    const dimOthers = asBoolean(named.dimOthers, true);
+
+    const api = getApiFromContext(ctx);
+    if (!api) {
+      // eslint-disable-next-line no-console
+      console.warn('[scrollyActions.highlightSeries] no api on parachart');
+      return;
+    }
+
+    const chart = api.getChart(ctx.chartId);
+    chart.highlightSeries(seriesIds, { emphasis, dimOthers });
+
+    // eslint-disable-next-line no-console
+    console.log('[highlightSeries]', {
+      chartId: ctx.chartId,
+      seriesIds,
+      emphasis,
+      dimOthers,
+    });
+  },
+
+  /**
+   * Reset highlight for a series or entire chart.
+   */
+  resetHighlight: (ctx: ActionContext, ...args: ParaActionArg[]) => {
+    const named = getNamedArgs(args);
+    const positional = getPositionalArgs(args);
+
+    const primarySeries = asString(named.series);
+    const seriesIds: string[] = [];
+
+    if (primarySeries) {
+      seriesIds.push(primarySeries);
+    }
+    for (const value of positional) {
+      const id = asString(value);
+      if (id) seriesIds.push(id);
+    }
+
+    const api = getApiFromContext(ctx);
+    if (!api || !ctx.chartId) {
+      // eslint-disable-next-line no-console
+      console.warn('[scrollyActions.resetHighlight] no api or chartId');
+      return;
+    }
+
+    const chart = api.getChart(ctx.chartId);
+
+    if (seriesIds.length > 0 && chart.resetHighlight) {
```

Comment:
Is there a case where `resetHighlight` or `resetAllHighlights` is not defined on `chart`?